pkg/client/block/client_integration_test.go

```diff
@@ -31,11 +31,17 @@ func TestBlockClient_LastNBlocks(t *testing.T) {
 func TestBlockClient_BlocksObservable(t *testing.T) {
 	ctx := context.Background()
 
-	blockClient := testblock.NewLocalnetClient(ctx, t)
+	blockClient, queryClient := testblock.NewLocalnetClient(ctx, t)
 	require.NotNil(t, blockClient)
 
 	blockSub := blockClient.CommittedBlocksSequence(ctx).Subscribe(ctx)
 
+	// Simulate the connection closing
+	go func() {
+		time.Sleep(1 * time.Second)
+		queryClient.Close()
```

Comment:
If the intended effect is to simulate a network failure and/or connection retry, I don't think this is the right way to go because while `eventsQueryClient#Close()` does close the underlying connection, it also calls `#UnsubscribeAll()` on the events bytes observable, which will effectively close the block client.

Reply:
Additionally, if this is the motivation behind the `testblock.NewLocalClient()` refactor, I would suggest reverting it if we decide not to do this here.

Reply:
I think this behavior should be tested in an unit test, not an integration test.

Reply:
This behavior might be more easily tested in a unit test where the `Connection` could be mocked and closed independently of the other components involved.

Reply:
I don't think we need to be testing for this case in an integration test.

Reply:
Agreed removing this and keeping it to unit tests